hedera-node/src/main/java/com/hedera/services/state/merkle/MerkleUniqueToken.java

```diff
@@ -108,41 +143,45 @@ public int getVersion() {
 
 	@Override
 	public void deserialize(SerializableDataInputStream in, int i) throws IOException {
-		owner = in.readSerializable();
-		creationTime = RichInstant.from(in);
+		ownerCode = in.readInt();
+		creationSecs = in.readLong();
+		creationNanos = in.readInt();
 		metadata = in.readByteArray(UPPER_BOUND_METADATA_BYTES);
 	}
 
 	@Override
 	public void serialize(SerializableDataOutputStream out) throws IOException {
-		out.writeSerializable(owner, true);
-		creationTime.serialize(out);
+		out.writeInt(ownerCode);
+		out.writeLong(creationSecs);
+		out.writeInt(creationNanos);
 		out.writeByteArray(metadata);
 	}
 
 	/* --- FastCopyable --- */
 	@Override
 	public MerkleUniqueToken copy() {
-		return new MerkleUniqueToken(owner, metadata, creationTime);
+		setImmutable(true);
+		return new MerkleUniqueToken(ownerCode, metadata, creationSecs, creationNanos);
 	}
 
 	public void setOwner(EntityId owner) {
-		this.owner = owner;
+		throwIfImmutable();
+		this.ownerCode = owner.identityCode();
 	}
 
 	public EntityId getOwner() {
-		return owner;
+		return new EntityId(0, 0, IdentityCodeUtils.numFromCode(ownerCode));
```

Comment:
Here there will be a cost in creating the `EntityId` object every time this is called, which can be a burden to GC. Also, depends on the usages (holding length) of the objects, it can take even more heap usage. Unlike before, it can return the referenced object and that reference can be repeatedly used.

Same for `getCreationTime()` call.

Reply:
These methods will only be used in answering queries, again creating short-lived, small objects that are very cheap to garbage-collect.

Reply:
Then this should be a big issue.